### ISSUE TITLE
`#{routing-suffix}` does not get populated

### ISSUE THREAD

**Issue by jstrachan** [minishift/minishift#1046]
here's a sample PR
https://gist.github.com/jstrachan/0d9bb130ffbb4f2f64316fb20d8f6daf

the routing-suffix does not get replaced with `nip.io` though - its blank https://github.com/jstrachan/minishift/blob/91814cf1403e5721931dd7ffc06cc74f7e44b8ff/addons/fabric8/fabric8.addon#L7

**Comment by hferentschik** [comment]
#{routing-prefix} is not a valid variable name. There is the #{routing-suffix} which gets interpolated. Is this what you want? And if not, what do you expect #{routing-prefix} to expand to?

**Comment by jstrachan** [comment]
sorry bad typing on the issue - I copy pasted the name `routing-suffix` from the docs into the source code and thats the one that doesn't work - I didn't copy the text into the issue though ;) just editted it

**Comment by jstrachan** [comment]
I just confirmed this error occurs on the default openshift 3.5.x as well as 3.6.x; it also doesn't matter if I deploy some Routes first before I try and use the `#{routing-suffix}` or not

**Comment by hferentschik** [comment]
@jstrachan Are you using `minishift addons apply fabric8` when you test this? My guess is yes. If you enable the plugin and run `minishift start` you most likely will see that the substitution will work. The problem is not so much that interpolation does not occur, but rather that the routing-suffix is not properly set when `addons apply` is executed.

Aside of this, I noticed something along the lines of `#{ip}.#{routing-suffix}` on the add-on. This way you will end up with the IP twice. The routing-suffix is not just 'nip.io' it is ip+'.nip.io'. So you only want to interpolate the routing suffix. 

I try to see whether there is an easy way to fix this.

**Comment by hferentschik** [comment]
@jstrachan FYI https://github.com/minishift/minishift/pull/1056

**Comment by jstrachan** [comment]
@hferentschik so previously I'd done something like this...
```
minishift start --vm-driver=xhyve --memory=7000 --cpus=4 --disk-size=50g 
minishift addons install --force addons/fabric8
minishift addons apply
```
after your comment I realised maybe the way you're meant to use addons is to install & enable them first then just run minishift start. So I tried that...
```
$ minishift addons install --force addons/fabric8
$ minishift addons enable fabric8
$ minishift start --vm-driver=xhyve --memory=7000 --cpus=4 --disk-size=50g --openshift-version=v3.6.0-alpha.1
Starting local OpenShift cluster using 'xhyve' hypervisor...
Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.0.2/minishift-b2d.iso'
 40.00 MiB / 40.00 MiB [===================================================================================================================================================================================================================================] 100.00% 0s
Downloading OpenShift binary 'oc' version 'v3.6.0-alpha.1'
 19.03 MiB / 19.03 MiB [===================================================================================================================================================================================================================================] 100.00% 0s
Starting OpenShift using openshift/origin:v3.6.0-alpha.1 ...
Pulling image openshift/origin:v3.6.0-alpha.1
Pulled 0/3 layers, 3% complete
Pulled 0/3 layers, 32% complete
Pulled 0/3 layers, 58% complete
Pulled 1/3 layers, 76% complete
Pulled 2/3 layers, 95% complete
Pulled 3/3 layers, 100% complete
Extracting
Image pull complete
OpenShift server started.

The server is accessible via web console at:
    https://192.168.64.82:8443

You are logged in as:
    User:     developer

To login as administrator:
    oc login -u system:admin

-- Applying addon 'fabric8':
  "Enabling CORS"..
  Installing the Fabric8 System using GitHub Client ID ${GITHUB_OAUTH_CLIENT_SECRET}.
  Adding the OAuthClient and cluster-admin role to the init-tenant service account..
  Please wait while the pods all startup!
  echo
  To watch this happening you can type:
    oc get pod -l provider=fabric8 -w
  echo
  Or you can watch in the OpenShift console via:
    minishift console
  echo
  When the pods are all running please click on the following URLs in your browser
  then ADVANCED, then click the URL at the bottom to approve the certs:
  echo
    https://keycloak-fabric8.192.168.64.82./
    https://wit-fabric8.192.168.64.82./api/status
    https://forge-fabric8.192.168.64.82./forge/version
    https://fabric8-fabric8.192.168.64.82./
  echo
  echo
  Then you should be able the open the fabric8 console here:
    https://fabric8-fabric8.192.168.64.82./
```
and still no `#{routing-suffix}` in the echo commands or the generated oauthclient
```
$ oc --as system:admin get oauthclient fabric8-online-platform
NAME                      SECRET    WWW-CHALLENGE   REDIRECT URIS
fabric8-online-platform   fabric8   FALSE           "https://keycloak-fabric8.192.168.64.82./auth/realms/fabric8/broker/openshift-v3/endpoint"
```
I wonder is there a different way to invoke the fabric8 addon where the `#{routing-suffix}` is applied?

**Comment by hferentschik** [comment]
> after your comment I realised maybe the way you're meant to use addons is to install & enable them first then just run minishift start. So I tried that...

There is no wrong or right. It depends what you are after. `addons apply` is great for development and testing of add-ons. Or when you create an add-on which does something ad-hoc, foe example you could have two add-ons which on demand (aka `addons apply`) toggle a certain feature.

The other use case is to configure OpenShift or deploy an application on each start of Minishift. That's where enabling add-ons comes in. Enabled add-ons will be executed on each creation of the VM (on the first start, not restart). So, for example, if my use case is to run Fabric8 with Minishift, I would most likely enable the fabirc8 add-on and get Fabric8 provision each time I start a new Minishift VM.

> I wonder is there a different way to invoke the fabric8 addon where the #{routing-suffix} is applied?

Since you were using `addons apply` your alternative is to `minishift addons enable fabric8 && minishift start`. This will indeed work, since on this code path the _routing-suffix_ will be properly set and interpolated. 

Regarding _routing-suffix_ not being interpolated on `addons apply` - I created a pull request #1056. I would expect it to be merged today and part of the release we are going to push coming Monday.

**Comment by jstrachan** [comment]
BTW I tried `minishift addons enable fabric8 && minishift start` as shown in the log above and that didn't work either

**Comment by hferentschik** [comment]
> BTW I tried minishift addons enable fabric8 && minishift start as shown in the log above and that didn't work either

Let me double check that.

**Comment by hferentschik** [comment]
> BTW I tried minishift addons enable fabric8 && minishift start as shown in the log above and that didn't work either

right, it was not working during `minishift start` either. Fixed now as well.

**Comment by hferentschik** [comment]
Merged via pull request #1056
